Xamarin.Forms.Core/Storyboard.cs

```diff
@@ -0,0 +1,551 @@
+			}
+			foreach (var anim in _animationsPerElement)
+			{
+				var elementAnimation = new Animation();
+				foreach (var animation in anim.Value)
+				{
+					elementAnimation.Add(0, 1, animation);
+
+				}
+				_runningAnimations.Add(elementAnimation);
+				elementAnimation.Commit(anim.Key, "ChildAnimations", 16, maxDuration, finished: (v, c) => Completed?.Invoke(this, new EventArgs()));
+			}
+		}
+
+		public new void Begin()
+		{
+			_animationsPerElement.Clear();
+			_runningAnimations.Clear();
+			var maxDuration = Duration;
+			var defaultTarget = Storyboard.GetTarget(this);
+			var defaultTargetProperty = Storyboard.GetTargetProperty(this);
+			foreach (var item in Children)
+			{
+				if (item is Animation<Color> timeline)
+				{
+					maxDuration = AddAnimationColor(maxDuration, defaultTarget, defaultTargetProperty, timeline);
+				}
+
+				if (item is DoubleAnimation doubleAnimation)
+				{
+					AddDoubleAnimation(defaultTarget, defaultTargetProperty, doubleAnimation);
+				}
+
+				if (item is PositionAnimation positionAnimation)
+				{
+					AddPositionAnimation(defaultTarget, defaultTargetProperty, positionAnimation);
+				}
+
+			}
+
+			foreach (var anim in _animationsPerElement)
+			{
+				var elementAnimation = new Animation();
+				foreach (var animation in anim.Value)
+				{
+					if (Reverse)
+					{
+						elementAnimation.Add(0, 0.5, animation);
+						elementAnimation.Add(0.5, 1, new Animation(animation.GetCallback(), 1, 0, Easing));
+					}
+					else
+					{
+						elementAnimation.Add(0, 1, animation);
+					}
+						
+				}
+				_runningAnimations.Add(elementAnimation);
+				elementAnimation.Commit(anim.Key, "ChildAnimations", 16, maxDuration,
+						finished: (v, c) => Completed?.Invoke(this, new EventArgs()),
+						repeat: () => Loop
+					);
+			}
+		}
+
+		void AddPositionAnimation(VisualElement defaultTarget, BindableProperty defaultTargetProperty, PositionAnimation positionAnimation)
+		{
+			var animationTarget = GetTarget(positionAnimation);
+			if (animationTarget == null)
+				animationTarget = defaultTarget;
+
+			var animationTargetProperty = GetTargetProperty(positionAnimation);
+			if (animationTargetProperty == null)
+				animationTargetProperty = defaultTargetProperty;
+
+
+			//maxDuration = Math.Max(maxDuration, doubleAnimation.Duration);
+			AddToInternalAnimationList(positionAnimation, animationTarget, animationTargetProperty);
+		}
+
+		void AddDoubleAnimation(VisualElement defaultTarget, BindableProperty defaultTargetProperty, DoubleAnimation doubleAnimation)
+		{
+			var animationTarget = Storyboard.GetTarget(doubleAnimation);
+			if (animationTarget == null)
+				animationTarget = defaultTarget;
+
+			var animationTargetProperty = Storyboard.GetTargetProperty(doubleAnimation);
+			if (animationTargetProperty == null)
+				animationTargetProperty = defaultTargetProperty;
+
+
+			//maxDuration = Math.Max(maxDuration, doubleAnimation.Duration);
```

Comment:
Comments..

Reply:
we need to implement this, I didn't had the logic to figure the maxDuration of the storyboard